src/utils/promise-to-iterable.ts

```diff
@@ -0,0 +1,32 @@
+import { $$asyncIterator } from 'iterall';
+
+export function createIterableFromPromise<T>(promise: Promise<T>): AsyncIterator<T> {
+  let isResolved = false;
+
+  return {
```

Comment:
@dotansimha since promise have only 1 result anyway,
shouldn't you return value, done: true on next()
and use return as teardown function (which doesn't do anything since promises are not cancelable)?